apps/gallery/test/marionette/lib/fakeactivitycaller.js

```diff
@@ -0,0 +1,83 @@
+/* global module */
+(function(module) {
+  'use strict';
+
+  var SELECTORS = Object.freeze({
+    pickMenu: 'form[data-z-index-level="action-menu"]',
+    pickImageButton: '#pick-image',
+    pickedImage: '#pick-activity-data > img',
+    pickedImageName: '#pick-activity-data > span.name',
+    pickedImageType: '#pick-activity-data > span.type',
+    sharedImage: '#share-activity-data > img',
+    sharedImageName: '#share-activity-data > span.name',
+    sharedImageType: '#share-activity-data > span.type'
+  });
+
+  var ORIGIN = 'gallery.fakeactivity.gaiamobile.org';
+
+  module.exports = {
+    ORIGIN: ORIGIN,
+
+    create: function(client) {
+      var originURL = 'app://' + ORIGIN;
+
+      return {
+        get pickMenu() {
+          // Switch to the system app first.
+          client.switchToFrame();
+          return client.helper.waitForElement(SELECTORS.pickMenu);
+        },
```

Comment:
Does the caller that uses this .pickMenu property have to remember to switch back to an app after accessing pickMenu?  That seems error-prone. Is there a way to wait for the element, then switch back before returning it?